From aecf6eee53a071a2816dbc47750c620bdfc64373 Mon Sep 17 00:00:00 2001
From: Aishwarya R <quic_aisr@quicinc.com>
Date: Tue, 1 Mar 2022 18:24:48 +0530
Subject: [PATCH] ath12k: don't call ath12k_pci_set_l1ss for WCN6855

For QCA6390, one PCI related clock drifts sometimes, and it makes
PCI link difficult to quit L1ss. Current implementation fixed this
by configuring some related regs using ath12k_pci_fix_l1ss.
WCN6855 does not have this clock drift problem, so no need to set
these regs.

Signed-off-by: Aishwarya R <quic_aisr@quicinc.com>
----
 drivers/net/wireless/ath/ath12k/core.c | 1 +
 drivers/net/wireless/ath/ath12k/hw.h   | 1 +
 drivers/net/wireless/ath/ath12k/pci.c  | 3 ++-
 3 files changed, 4 insertions(+), 1 deletion(-)
--- a/drivers/net/wireless/ath/ath12k/hw.h
+++ b/drivers/net/wireless/ath/ath12k/hw.h
@@ -190,6 +190,8 @@ struct ath12k_hw_params {
 
 	const char *qdss_config;
 	unsigned int qdss_config_len;
+
+	bool fix_l1ss;
 };
 
 /* BRINGUP: move to dp.h */
--- a/drivers/net/wireless/ath/ath12k/pci.c
+++ b/drivers/net/wireless/ath/ath12k/pci.c
@@ -301,6 +301,9 @@ static void ath12k_pci_sw_reset(struct a
 		ath12k_pci_enable_ltssm(ab);
 		ath12k_pci_clear_all_intrs(ab);
 		ath12k_pci_set_wlaon_pwr_ctrl(ab);
+
+		if (ab->hw_params->fix_l1ss)
+			ath12k_dbg(ab, ATH12K_DBG_PCI, "L1ss fix required\n");
 	}
 
 	ath12k_mhi_clear_vector(ab);
--- a/drivers/net/wireless/ath/ath12k/hw.c
+++ b/drivers/net/wireless/ath/ath12k/hw.c
@@ -875,6 +875,8 @@ static const struct ath12k_hw_params ath
 
 		.qdss_config = NULL,
 		.qdss_config_len = 0,
+
+		.fix_l1ss = false,
 	},
 	{
 		.name = "wcn7850 hw2.0",
@@ -934,6 +936,8 @@ static const struct ath12k_hw_params ath
 
 		.qdss_config = ath12k_hw_qdss_config_wcn7850,
 		.qdss_config_len = sizeof(ath12k_hw_qdss_config_wcn7850),
+
+		.fix_l1ss = false,
 	},
 };
 
